portals/consent-portal/src/ConsentContext.tsx

```
@@ -0,0 +1,183 @@
+}
+
+const ConsentContext = createContext<ConsentContextType | undefined>(undefined);
+
+export const useConsent = () => {
+  const context = useContext(ConsentContext);
+  if (!context) {
+    throw new Error('useConsent must be used within ConsentProvider');
+  }
+  return context;
+};
+
+export const ConsentProvider: React.FC<{ children: ReactNode }> = ({children}) => {
+  const navigate = useNavigate();
+  const [consentRecord, setConsentRecord] = useState<ConsentRecord | null>(null);
+  const [error, setError] = useState('');
+  const [consentId, setConsentId] = useState<string | null>(null);
+  const [userEmail, setUserEmail] = useState<string>('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const {isSignedIn, user, getAccessToken} = useAsgardeo();
+
+  const CONSENT_ENGINE_PATH = window?.configs?.apiUrl;
+
+  useEffect(() => {
+    if (!CONSENT_ENGINE_PATH) {
+      setError('API URL is not configured properly.');
+      navigate('/error');
+    }
+  }, [CONSENT_ENGINE_PATH, navigate]);
+
+  const getAuthHeaders = useCallback(async (): Promise<HeadersInit> => {
+    const accessToken = await getAccessToken();
+    const headers: HeadersInit = {
+      'Content-Type': 'application/json',
+    };
+
+    if (accessToken) {
+      headers['Authorization'] = `Bearer ${accessToken}`;
+    }
+
+    return headers;
+  }, [getAccessToken]);
+
+  const fetchUserInfo = useCallback(() => {
+    if (user) {
+      setUserEmail(user.email || '');
+    }
+  }, [user]);
+
+  const fetchConsentData = useCallback(async (consentUuid: string) => {
+    try {
+      const headers = await getAuthHeaders();
+      const response = await fetch(`${CONSENT_ENGINE_PATH}/consents/${consentUuid}`, {headers});
+
+      if (!response.ok) {
+        let errorMessage = '';
+        try {
+          const errorData = await response.json();
+          errorMessage = errorData.error || '';
+        } catch { /* empty */
```

why error handling in `fetchConsentData` and `handleConsentDecision` not properly handle all error cases? some errors may be silently ignored.

suggest

```typescript
let errorMessage = '';
try {
  const errorData = await response.json();
  errorMessage = errorData.error || errorData.message || `HTTP ${response.status}`;
} catch (parseError) {
  // If JSON parsing fails, use status text
  errorMessage = response.statusText || `HTTP ${response.status}`;
  console.warn('Failed to parse error response:', parseError);
}
```